pkg/service/psetupsvc/complete_step.go

```diff
@@ -66,7 +66,9 @@ func (s *Service) CompleteStep(ctx context.Context, setupID string, stepIndex in
 	}
 
 	// load the current values into this config
-	err = cfg.Load(ctx, &gconfig.MapLoader{Values: setup.ConfigValues})
+	// Skip secrets is set to true because we never need to read the value of a secret in this context.
+	// If a value is provided for a secret it will be updated
+	err = cfg.Load(ctx, &gconfig.MapLoader{Values: setup.ConfigValues, SkipLoadingSecrets: true})
```

Comment:
I found a bug in the provider setup UI where config including a secret value failed in teh deployed version because the GetParameter IAM permission is missing from the API server for provider secrets.

What I decided is that in the context of configuration, we never need to retrieve the value of the secret from SSM, so instead, I have added a flag to the config loaders which skips loading secrets as an opt in.

This now works as expected, in the UI you can update any config value without it failing with a permission error, and we don't need to have read access to provider secrets in the client facing API.